metriq_gym/benchmarks/benchmark.py

```diff
@@ -1,17 +1,31 @@
 import argparse
+from collections.abc import Iterable
 
 from pydantic import BaseModel
 from dataclasses import dataclass
 
 from qbraid import GateModelResultData, QuantumDevice, QuantumJob
 
 
+def flatten_job_ids(quantum_job: QuantumJob | Iterable[QuantumJob]) -> list[str]:
+    if isinstance(quantum_job, QuantumJob):
+        return [quantum_job.id]
+    elif isinstance(quantum_job, Iterable):
+        return [job.id for job in quantum_job]
+    else:
+        raise TypeError(f"Unsupported job type: {type(quantum_job)}")
+
+
 @dataclass
 class BenchmarkData:
     """Stores intermediate data from pre-processing and dispatching"""
 
     provider_job_ids: list[str]
 
+    @classmethod
+    def from_quantum_job(cls, quantum_job, **kwargs):
```

Comment:
[nit]: Should we include a return type as well here as well as type-hinting for the args?